developer/src/kmc/src/util/NodeCompilerCallbacks.ts

```diff
@@ -154,6 +158,25 @@ export class NodeCompilerCallbacks implements CompilerCallbacks {
       // Reset max message limit when a new file is being processed
       this.messageFilename = event.filename;
       this.messageCount = 0;
+      this.messageFiletext = '';
+    }
+
+    if (event.column && !event.line && event.filename) {
+      try {
+        let text = '';
+        if (this.messageFilename == event.filename) {
+          text = this.messageFiletext;
+        }
+        if (text == '') {
+          text = this.fs.readFileSync(event.filename, "utf-8");
+          if (this.messageFilename == event.filename) {
+            this.messageFiletext = text;
+          }
```

Comment:
The message processor should not attempt to load source text from disk. It may not have come from disk -- it may be a network resource, memory resource, etc.

Reply:
so the call site down in keys.ts etc doesn't seem to have access to the source XML - maybe it should? 

I think the challenge i see is that the callbacks with reportMessage goes 'directly' back to the consumer, the outer compiler.compile() function (which has the XML source) doesn't get involved.

Reply:
think i've resolved this..